README.md

```diff
@@ -1,634 +1,1855 @@
-
-3. **Borrower makes a payment** - The borrower cannot make payments once a loan has expired. This is enforced by the borrower having to use `invalid-hereafter` to specify the current time. Since the borrower wants as much time as possible to pay off the loan, but *must* do so prior to expiration slot, **they are incentivized to set `invalid-hereafter` as close the expiration slot as possible.**
-
-#### Interest
-Unlike other lending/borrowering protocols, Cardano-Loans does not use an algorithm to determine the interest rates. Instead, rates are one of the explicitly negotiated terms between borrowers and lenders.
-
-**Cardano-Loans v1 utilizes fixed, non-compounding interest rates.** Validity intervals cannot tell the current time during payments, they can only tell whether a certain time has passed or not. Compounding requires the state of the loan to be periodically updated, which is possible by breaking the loan up into "epochs", using `invalid-hereafter` to check whether the current loan-epoch has passed. This is similar to the expiry check that occurs when the borrower makes a repayment, though it adds complexity and logic to the design. For a more detailed explanation, see the [Future Directions](#future-directions-and-considerations) section.
-
-Since Cardano-Loans v1 is non-compounding, the total amount owed for all loans is always:
-```
-amountOwed = principle * (1 + interestRate)
+This section is the low-level specification for the protocol. If you are only interested in the
+high-level aspects, feel free to skip to the next [section](#benchmarks-and-fee-estimations-ymmv).
+
+### The Borrower's Address
+
+Borrowers each create a unique "loan" address - this is where all negotiations take place *and*
+where all collateral is kept until the loan is either repaid or expired. As is common in
+*distributed dApps*, all such loan addresses use the same validator script for the payment
+credential, and a unique, user-defined staking key or script for the staking credential.
+Owner-related actions are delegated to the staking credential *by* the validator script, so the user
+maintains full control of all assets at the address. The borrower's credit history is
+cryptographically tied to the staking credential used in the loan address.
+
+Since negotiations occur in the borrower's address, the borrower maintains staking and voting rights
+over all assets throughout the life cycle of the loan. This includes collateral, which makes sense,
+since collateral belongs to the borrower until they default. It also includes any assets attached to
+offers, which incentivizes lenders to be proactive and not leave offers open for too long. 
```

Comment:
Does this mean it is possible for a potential borrower to create an address, deposit some Ada to it and continue to receive staking rewards (so no ongoing cost).  Then simply wait for loan offers but not respond, and benefit from the increased staking yield when potential lenders send stake-able tokens as a loan offer?

Reply:
Yes, but I don't see this as an issue for two reason:
1. Ask UTxOs are timestamped. If this attack is actually an issue, lenders can just avoid old Ask UTxOs.
2. The Cardano epoch is 5 days. If a borrower is *actually* looking for a loan, why would it take 5 days to accept a lender's offer? Besides, if lenders want to protect their staking rewards, they can easily close all of their offers right before the end of the epoch and open them again after the next snapshot. In this scenario, the borrower wouldn't benefit at all from this attack. Front-ends can also remind lenders to do this.